geopandas/geodataframe.py

```diff
@@ -274,6 +274,8 @@ def set_geometry(self, col, drop=False, inplace=False, crs=None):
 
         to_remove = None
         geo_column_name = self._geometry_column_name
+        if isinstance(self.columns, pd.MultiIndex) and isinstance(geo_column_name, str):
+            geo_column_name = (geo_column_name, "")
```

Comment:
This solution needs to be generalisable to `n` levels of the MutliIndex. This fix assumes 2 levels but in reality, there can be more than that.

Reply:
Well that should have been obvious to me at the time 😅 . What's curious is that if I modify the test I added to have a 3 level multiindex, this change isn't needed at all:
```python
# GH1763 https://github.com/geopandas/geopandas/issues/1763
df = pd.DataFrame(
    # [[1, 0], [0, 1]], columns=[["location", "location"], ["x", "y"]]
    [[1, 0], [0, 1]], columns=[['foo', 'foo'], ["location", "location"], ["x", "y"]]
)
x_col = df["foo", "location", "x"]
y_col = df["foo", "location", "y"]
print(df.columns)
gdf = GeoDataFrame(df, crs=self.wgs, geometry=points_from_xy(x_col, y_col))
print(gdf.columns)
assert gdf.crs == self.wgs

gdf = GeoDataFrame(df, geometry=points_from_xy(x_col, y_col, crs=self.wgs))
assert gdf.crs == self.wgs
```.

So something special must be going on in the 2 level multiindex case I guess? Still need to look.